compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1852,8 +1854,107 @@ setRootConfig(mlir::FunctionOpInterface entryPointFn,
                              vecPreProcStrategy);
 }
 
+// Utility function to return the result inner tile sizes associated with the
+// mmt4d op. I.e. returns the M1 and N1 dimensions of the result shape
+// BxM0xN0xM1xN1. In case of scalable dimensions, walks the IR and returns
+// the static inner tile size and the appropriate scalable flag.
+// TODO: By using the IterationDimTracker, extend the logic to
+// infer all inner tile sizes - including the inner K dimension, if possible.
+static FailureOr<SizesAndScalableFlags>
+getMmt4dInnerTileSizes(linalg::LinalgOp op) {
+  Value destValue = op.getDpsInitOperand(0)->get();
+  ArrayRef<int64_t> destShape =
+      cast<ShapedType>(destValue.getType()).getShape();
+  // In case of static tiles, we just return the last two dim sizes.
+  if (ShapedType::isStaticShape(destShape)) {
+    return SizesAndScalableFlags{destShape.take_back(2),
+                                 IREE::Codegen::ScalableTileFlags(2, false)};
+  }
+  Operation *destOp = destValue.getDefiningOp();
+  SmallVector<OpFoldResult> innerDims;
+  // Currently, the only init tensor of the mmt4d seems to be linalg.fill ops
+  // or the bindings. Otherwise, bail out.
+  if (!isa<linalg::FillOp, IREE::TensorExt::DispatchTensorLoadOp>(destOp)) {
```

Comment:
The better solution to me is implementing `ShapeAwareOpInterface` for `tensor.empty` op and track the list of `linalg.fill` until it reaches the interface op, but I'm unsure if it opens a can of worms or not. I'll try to implement it and see if it works.